armi/reactor/components/component.py

```diff
@@ -226,6 +226,8 @@ class Component(composites.Composite, metaclass=ComponentType):
 
     pDefs = componentParameters.getComponentParameterDefinitions()
 
+    material: materials.Material
```

Comment:
I feel like this will confuse people with the import at the top of this file:

> from armi.materials import material

At the very least, this variable needs to be called `materialClass`.

But, probably it would be cleaner not to save class types as class attributes. Just do `isinstance(Material)` directly, it will be easier to read.

Reply:
There is no assignment nor variable creation; there's no `=` sign. It's just a type hint that every attribute *should* be of type `materials.Material`.
```python
In [1]: class D:
   ...:     x: int
   ...:

In [2]: d = D()

In [3]: d.x
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[3], line 1
----> 1 d.x

AttributeError: 'D' object has no attribute 'x'
```
https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#classes

Reply:
Yeah, I see that is a colon, I _read_ it as an equal sign.

And then I probably got excited and just slapped a "Mission Accomplished" banner on that aircraft carrier and moved on.

I see it now.